libs/test/SmartSyncTest/src/com/salesforce/androidsdk/smartsync/target/ParentChildrenSyncTest.java

```diff
@@ -701,89 +678,800 @@ public void testCleanResyncGhostsForParentChildrenTarget() throws Exception {
 
         // Checking db
         checkDb(accountIdToFieldsLeft, ACCOUNTS_SOUP);
-        checkDbDeleted(ACCOUNTS_SOUP, new String[] {accountIdDeleted}, Constants.ID);
+        checkDbDeleted(ACCOUNTS_SOUP, new String[]{accountIdDeleted}, Constants.ID);
         for (String accountId : accountIdContactIdToFields.keySet()) {
             if (accountId.equals(accountIdDeleted)) {
                 checkDbDeleted(CONTACTS_SOUP, accountIdContactIdToFields.get(accountId).keySet().toArray(new String[0]), Constants.ID);
-            }
-            else {
+            } else {
                 checkDb(accountIdContactIdToFields.get(accountId), CONTACTS_SOUP);
 
             }
         }
     }
 
+    /**
+     * Create accounts and contacts locally, sync up with merge mode OVERWRITE, check smartstore and server afterwards
+     */
+    public void testSyncUpWithLocallyCreatedRecords() throws Exception {
+        trySyncUpWithLocallyCreatedRecords(SyncState.MergeMode.OVERWRITE);
+    }
 
     /**
-     * Helper method for the testDelete*
-     * @param relationshipType
-     * @param singleDelete
-     * @throws JSONException
+     * Create accounts and contacts locally, sync up with mege mode LEAVE_IF_CHANGED, check smartstore and server afterwards
      */
-    protected void tryDeleteFromLocalStore(ParentChildrenSyncDownTarget.RelationshipType relationshipType, boolean singleDelete) throws JSONException {
-        String[] accountNames = {
-                createRecordName(Constants.ACCOUNT),
-                createRecordName(Constants.ACCOUNT),
-                createRecordName(Constants.ACCOUNT)
-        };
-        Map<JSONObject, JSONObject[]> mapAccountToContacts = createAccountsAndContactsLocally(accountNames, 3);
-        JSONObject[] accounts = mapAccountToContacts.keySet().toArray(new JSONObject[] {});
+    public void testSyncUpWithLocallyCreatedRecordsWithoutOverwrite() throws Exception {
+        trySyncUpWithLocallyCreatedRecords(SyncState.MergeMode.LEAVE_IF_CHANGED);
+    }
 
-        String[] contactIdsOfFirstAccount = JSONObjectHelper.pluck(mapAccountToContacts.get(accounts[0]), Constants.ID).toArray(new String[0]);
-        String[] contactIdsOfSecondAccount = JSONObjectHelper.pluck(mapAccountToContacts.get(accounts[1]), Constants.ID).toArray(new String[0]);
-        String[] contactIdsOfThirdAccount = JSONObjectHelper.pluck(mapAccountToContacts.get(accounts[2]), Constants.ID).toArray(new String[0]);
+    /**
+     * Sync up with locally updated child
+     *
+     * @throws Exception
+     */
+    public void testSyncUpLocallyUpdatedChild() throws Exception {
```

Comment:
The next 38 tests try all meaningful permutation of local and/or remote update/delete of parent and/or child. 
trySyncUpsWithVariousChanges does all the work.